### DISCUSSION TITLE
calculator break down

### DISCUSSION THREAD

**Discussion by jasper12138** [SSCHAcode/python-sscha#132] [Q&A] [answered]
Hi developers,
when I run sscha, task may break down sometimes with output below:

##############
Computing configuration 77 out of 200 (nat = 64)
Computing configuration 78 out of 200 (nat = 64)
Computing configuration 79 out of 200 (nat = 64)
Rerun the job 78
Rerun the job 78
Rerun the job 78
Rerun the job 78
Rerun the job 78
###############

In the error report, it says:
###########
srun: Job xxxxxx step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Step created for job xxxxxx
###########

I used 4 nodes for sscha.py script in this task.  

Also I do not know how to improve the performance of sscha+qe.  When I use 4 nodes and 1 nodes for the same sscha.py script, the job running time was almost the same (I detected the running time in espresso.out).

Best wishes.

**Comment by diegomartinez2** [comment, marked as answer]
Hi, what are you using, sbatch or srun? What is the command/script you use on the cluster?
Maybe without  ”--mpi=pmi2” it will work? I need more information.

**Reply by jasper12138** [reply]
Hi, I work on a slurm system. I submitted jobs by sbatch with mpi parallelization with srun as below:

#########
#!/bin/bash
#SBATCH --job-name=sscha
#SBATCH --output=%j.out
#SBATCH --error=%j.err
#SBATCH --ntasks=48
#SBATCH --cpus-per-task=4

python sscha.py
########

now I updated the command in sscha.py:
old: cmd = "srun pw.x -nk 48 -in PREFIX.pwi > PREFIX.pwo "
new: cmd = "srun pw.x -nk 48 -in PREFIX.pwi > PREFIX.pwo ; sleep 5"
it runs smoothly now without any warning or error for one generation of 200 configurations.
I guess the former error reports are caused by conflicts in task sequences.
In the other words, one configuration scf calculation do not end, but the latter one has started.

**Comment by mesonepigreco** [comment]
Hi @jasper12138 ,
As Diego is pointing out, ts a be a problem with the scheduler of your cluster.
An alternative would be to use the Cluster submission module from a client machine that manages the submission on the cluster. However, this depends if you have access to a local workstation and a simple way to connect to the cluster.

**Reply by jasper12138** [reply]
Hi, I worked on the cluster with only static password. It is possible to do it as you said. I will try it.
